src/solc_0.7/catalyst/AssetMinter.sol

```diff
@@ -112,7 +112,7 @@ contract AssetMinter is WithMetaTransaction {
 
         for (uint256 i = 0; i < assets.length; i++) {
             if (assets[i].catalystId != 0) {
-                require(catalystsQuantities[assets[i].catalystId] != 0, "INVALID_CATALYST_NOT_ENOUGH");
+                require(catalystsQuantities[i] != 0, "INVALID_CATALYST_NOT_ENOUGH");
```

Comment:
Previously, this was `require(catalystsQuantities[assets[i].catalystId] != 0,...`
- `assets[i].catalystId` evaluates to a number (the value for the `catalystId` field for the asset at index `i` in the `assets` array)
- this number is then passed as the index to lookup in the `catalystsQuantities` array.
- as `catalystId` should **never** be 0, we would therefore never look at the value at index `0` in `catalystsQuantities`

Reply:
Using `i` might be a good idea, as it may save some gas, (reading from stack instead of memory), and I'm not sure what Ronan meant, but, you should have a look at how gems and catalysts are stored in the `GemsCatalystsRegistry` contract as an example(specifically `getGem/getCatalyst`).

Reply:
Hmm. The more I look at this, the more I think that the `catalystQuantities` param must be of this form:
`[0, amountOfId1, amountOfId2, amountOfId3, amountOfId4]`, where the 0-index is padded to offset the values to the right by 1.

Reply:
not a bug.